tests/server/chunk_server/chunk_server_file_service_test.cc

```diff
@@ -0,0 +1,187 @@
+  EXPECT_TRUE(metadata_manager->AdvanceChunkVersion("0").ok());
+  // Set the chunk server location to be the one used in this test
+  auto file_chunk_metadata(metadata_manager->GetFileChunkMetadata("0")
+                               .ValueOrDie());
+  file_chunk_metadata.mutable_locations()->Add(
+      tests::ChunkServerLocationBuilder(kTestChunkServerHostName,
+                                        kTestChunkServerPort));
+  metadata_manager->SetFileChunkMetadata(file_chunk_metadata);
+}
+
+// Start a test chunk server, this gets called from a separate thread
+void StartTestChunkServer() {
+  ServerBuilder builder;
+  auto credentials = grpc::InsecureServerCredentials();
+  builder.AddListeningPort(kTestChunkServerAddress, credentials);
+  // Register the chunk server file service impl
+  StatusOr<ChunkServerImpl*> chunk_server_or =
+      ChunkServerImpl::ConstructChunkServerImpl(kTestConfigPath,
+                                                kTestChunkServerName);
+  ChunkServerImpl* chunk_server = chunk_server_or.ValueOrDie();
+  ChunkServerFileServiceImpl file_service(chunk_server);
+  builder.RegisterService(&file_service);
+  // Initialize test data on the chunk server
+  SeedTestChunkData(); 
+  // Start the server, and let it run until thread is cancelled
+  std::unique_ptr<Server> server(builder.BuildAndStart());
+  server->Wait();
+}
+
+// Start a test master server, this gets called from a separate thread
+void StartTestMasterServer() {
+  ServerBuilder builder;
+  auto credentials = grpc::InsecureServerCredentials();
+  builder.AddListeningPort(kTestMasterServerAddress, credentials);
+  // Register the master metadata service impl
+  ConfigManager* config = ConfigManager::GetConfig(kTestConfigPath)
+                              .ValueOrDie();
+  MasterMetadataServiceImpl metadata_service(config);
+  builder.RegisterService(&metadata_service);
+  // Initialize test data on the master server
+  SeedTestMetadata();
+  // Start the server, and let it run until thread is cancelled
+  std::unique_ptr<Server> server(builder.BuildAndStart());
+  server->Wait();
+}
+
+// Initialize the client impl and make it ready to call client function
+void StartClient() {
+  Status init_status(gfs::client::init_client(kTestConfigPath, 
+                                              "master_server_01",true));
+  EXPECT_TRUE(init_status.ok());
+}
+
+void SingleClientReadShortData() {
+  auto read_result(gfs::client::read(short_data_filename.c_str(), 
+                                     0, short_data.size()));
+  if (!read_result.ok()) {
+    std::cerr << read_result.status().error_message() << std::endl;
+  }
+  EXPECT_TRUE(read_result.ok());
+  // Make sure that the read data is expected
+  auto read_data(read_result.ValueOrDie());
+  EXPECT_EQ(read_data.bytes_read, short_data.size());
+  EXPECT_EQ(memcmp(read_data.buffer, short_data.c_str(), short_data.size()), 0);
+}
+
+class ChunkServerFileServiceTest : public ::testing::Test {
```

Comment:
I am implementing this, so please add TODO and assign it to me, so I can refactor/rewrite this test for more cases. Won't read it now